specs/ProgrammaticSaveAs.md

```diff
@@ -0,0 +1,429 @@
+  ///
+  /// \snippet ScenarioSaveAs.cpp ProgrammaticSaveAs
+  HRESULT SaveContentAs([in] ICoreWebView2RequestSaveAsCompletedHandler* handler);
+
+  /// Add an event handler for the `SaveAsRequested` event. This event is raised
+  /// when save as is triggered, programmatically or manually.
+  ///
+  /// \snippet ScenarioSaveAs.cpp ToggleEventHandler
+  HRESULT add_SaveAsRequested(
+   [in] ICoreWebView2SaveAsRequestedEventHandler* eventHanlder,
+   [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_SaveAsRequested`.
+  ///
+  /// \snippet ScenarioSaveAs.cpp ToggleEventHandler
+  HRESULT remove_SaveAsRequested(
+   [in] EventRegistrationToken token);
+}
+
+/// The event handler for the `SaveAsRequested` event.
+[uuid(55b86cd2-adfd-47f1-9cef-cdfb8c414ed3), object, pointer_default(unique)]
+interface ICoreWebView2SaveAsRequestedEventHandler : IUnknown {
+  HRESULT Invoke(
+   [in] ICoreWebView2* sender,
+   [in] ICoreWebView2SaveAsRequestedEventArgs* args);
+}
+
+/// The event args for `SaveAsRequested` event
+[uuid(80101027-b8c3-49a1-a052-9ea4bd63ba47), object, pointer_default(unique)]
+interface ICoreWebView2SaveAsRequestedEventArgs : IUnknown {
+  /// Get the Mime type of content to be saved
+  [propget] HRESULT ContentMimeType([out, retval] LPWSTR* value);
+
+  /// Indicates if client side cancelled the silent save as, TRUE means cancelled.
+  /// When the event is invoked, the download won't start. A programmatic call will
+  /// return COREWEBVIEW2_SAVE_AS_CANCELLED as well. 
+  ///
+  /// The default value is FALSE.
+  ///
+  /// Set the `Cancel` for save as
+  [propput] HRESULT Cancel ([in] BOOL value);
+
+  /// Get the `Cancel` for save as
+  [propget] HRESULT Cancel ([out, retval] BOOL* value);
+
+  /// Indicates if the system default dialog will be supressed, FALSE means
+  /// save as default dialog will show; TRUE means a silent save as, will
+  /// skip the system dialog. 
+  ///
+  /// The default value is FALSE.
+  ///
+  /// Set the `SupressDefaultDialog`
+  [propput] HRESULT SupressDefaultDialog([in] BOOL value);
+
+  /// Get the `SupressDefaultDialog`
+  [propget] HRESULT SupressDefaultDialog([out, retval] BOOL* value);
+
+  /// Returns an `ICoreWebView2Deferral` object. This will defer showing the 
+  /// default Save As dialog and performing the Save As operation.
+  HRESULT GetDeferral([out, retval] ICoreWebView2Deferral** deferral);
+
+  /// `ResultFilePath` is absolute full path of the location. It includes the
+  /// file name and extension. If `ResultFilePath` is not valid, e.g. root drive
+  /// not exist, save as will be denied and return COREWEBVIEW2_SAVE_AS_INVALID_PATH.
+  ///
+  /// When the download complete and success, a target file will be saved at this
+  /// location. If the select `COREWEBVIEW2_SAVE_AS_REQUESTED_KIND_COMPLETE`, will be 
+  /// an additional directory with resources files. The directory has the same name 
```

Comment:
It cannot have the same name at the same location. The name must be different somehow. Is it the same filename but without a file extension?